.github/workflows/turing.yaml

```diff
@@ -354,175 +352,137 @@ jobs:
+          kubectl logs deployment/turing-merlin
           echo "::endgroup::"
 
-          tee credentials.json <<EOF
-          {
-            "master_ip": "kubernetes:443",
-            "certs": "$(k3d kubeconfig get ${{ env.CLUSTER_NAME }} | yq e '.clusters[0].cluster.certificate-authority-data' - | base64 --decode | awk 1 ORS='\\n')",
-            "client_certificate": "$(k3d kubeconfig get ${{ env.CLUSTER_NAME }} | yq e '.users[0].user.client-certificate-data' - | base64 --decode | awk 1 ORS='\\n')",
-            "client_key": "$(k3d kubeconfig get ${{ env.CLUSTER_NAME }} | yq e '.users[0].user.client-key-data' - | base64 --decode | awk 1 ORS='\\n')"
-          }
-          EOF
-
-          curl -v \
-            --header "X-Vault-Token: ${{ env.VAULT_TOKEN }}" \
-            --request POST \
-            --data @credentials.json \
-            http://vault.127.0.0.1.nip.io/v1/secret/dev
-
-          # Create a secret for Merlin with information on how to access Vault
-          kubectl create secret generic vault-secret \
-            --from-literal=vault-address=http://vault:8200 \
-            --from-literal=vault-token=${{ env.VAULT_TOKEN }}
-
-      - name: Install Turing
-        run: |
-          helm install turing ./infra/charts/turing \
-            --values infra/e2e/turing.values.yaml \
-            --values infra/e2e/${{ matrix.valuesFile }} \
-            --set turing.image.registry=${{ env.LOCAL_REGISTRY }} \
-            --set turing.image.repository=${{ github.repository }}/turing-api \
-            --set turing.image.tag=${{ env.TURING_API_VERSION }} \
-            --set turing.config.RouterDefaults.Image=${{ env.LOCAL_REGISTRY }}/${{ github.repository }}/turing-router:${{ env.TURING_ROUTER_VERSION }} \
-            --set turing.experimentEngines[0].rpcPlugin.image=${{ env.LOCAL_REGISTRY }}/${{ github.repository }}/test-experiment-engine-plugin:${{ env.TEST_EXPERIMENT_ENGINE_PLUGIN_VERSION }}
-
-      - name: Install mockserver
-        run: |
-          kubectl apply -f infra/e2e/turing.mockserver.yaml
+          echo "::group::logs deployment/turing"
+          kubectl logs deployment/turing
+          echo "::endgroup::"
+
+  test-e2e-sdk:
+    runs-on: ubuntu-latest
+    env:
+      CLUSTER_INIT_VERSION: ${{ needs.build-cluster-init.outputs.cluster-init-version }}
+      CLUSTER_NAME: turing-e2e
+      ISTIO_VERSION: 1.9.9
+      KNATIVE_VERSION: v0.18.3
+      KNATIVE_ISTIO_VERSION: v0.18.1
+      LOCAL_REGISTRY: registry.localhost:5000
+      TURING_API_VERSION: ${{ needs.build-api.outputs.api-version }}
+      TURING_ROUTER_VERSION: ${{ needs.build-router.outputs.router-version }}
+      TEST_EXPERIMENT_ENGINE_PLUGIN_VERSION: ${{ needs.build-test-experiment-engine-plugin.outputs.test-experiment-engine-plugin-version }}
+    needs:
+      - build-api
+      - build-router
+      - build-test-experiment-engine-plugin
+      - build-cluster-init
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
This matrix effectively produces 6 configurations. While it's helpful to be exhaustive, since the e2e tests are heavy (~10 mins for each job), it could be wasteful to run all 6 configs with every PR, unless they are expected to produce different results (eg: does `useInClusterConfig` matter for the SDK <> API tests? What about the Python version?). Besides, we also have 2 jobs for the API e2e already.

Given these, do you have any ideas on how we can reduce the test combinations?

Reply:
Hmm you're right, `useInClusterConfig` and the python versions shouldn't drastically affect the results of the e2e tests. I'll remove the redundant configurations then and leave the sdk e2e tests to solely run on remote cluster credentials and python 3.9. I'm not sure if we're aiming to implement multiple sets of e2e tests for each release of the SDK (but I doubt so) so I'm leaving that out too.